sources/packages/api/src/route-controllers/education-program/education-program.controller.ts

```diff
@@ -35,27 +47,42 @@ export class EducationProgramController {
     private readonly formService: FormService,
   ) {}
 
+  /**
+   * Get programs for a particular institution with pagination.
+   * @param locationId id of the location.
+   * @param pageSize is the number of rows shown in the table
+   * @param page is the number of rows that is skipped/offset from the total list.
+   * For example page 2 the skip would be 10 when we select 10 rows per page.
+   * @param sortField the sorting column.
+   * @param sortOrder sorting order.
+   * @param searchProgramName Search the program name in the query
+   * @returns EducationProgramsSummaryPaginated.
+   */
   @AllowAuthorizedParty(AuthorizedParties.institution)
   @HasLocationAccess("locationId")
   @Get("location/:locationId/summary")
   async getSummary(
     @Param("locationId") locationId: number,
+    @Query("searchProgramName") searchProgramName: string,
+    @Query("sortField") sortField: string,
+    @Query("sortOrder") sortOrder: FieldSortOrder,
     @UserToken() userToken: IInstitutionUserToken,
-  ): Promise<SummaryEducationProgramDto[]> {
-    const programs = await this.programService.getSummaryForLocation(
+    @Query("page") page = DEFAULT_PAGE_NUMBER,
+    @Query("pageLimit") pageLimit = DEFAULT_PAGE_LIMIT,
+  ): Promise<EducationProgramsSummaryPaginated> {
+    // [OfferingTypes.applicationSpecific] offerings are created during PIR, if required
```

Comment:
Can this comment be expanded? 
// [OfferingTypes.applicationSpecific] offerings are created during PIR, if required, and they are supposed to be viewed only associated to the application that they were associated to during the PIR, hence they should not be displayed alongside with the public offerings.